app/models/topic_converter.rb

```diff
@@ -66,18 +66,45 @@ def posters
     @posters ||= @topic.posts.where("post_number > 1").distinct.pluck(:user_id)
   end
 
+  def increment_users_post_count
+    update_users_post_count(:increment)
+  end
+
+  def decrement_users_post_count
+    update_users_post_count(:decrement)
+  end
+
+  def update_users_post_count(action)
+    operation = action == :increment ? "+" : "-"
+
+    DB.exec(<<~SQL)
+    UPDATE user_stats
+    SET post_count = post_count #{operation} X.count
+    FROM (
+      SELECT
+        us.user_id,
+        COUNT(*) AS count
+      FROM user_stats us
+      INNER JOIN posts ON posts.topic_id = #{@topic.id.to_i} AND posts.user_id = us.user_id
+      WHERE posts.post_number > 1
+      AND NOT posts.hidden
+      AND posts.post_type = #{Post.types[:regular].to_i}
+      GROUP BY us.user_id
+    ) X
+    WHERE X.user_id = user_stats.user_id
+    SQL
+  end
+
   def update_user_stats
-    # update posts count. NOTE that DirectoryItem.refresh will overwrite this by counting UserAction records.
-    # update topics count
-    UserStat.where(user_id: posters).update_all('post_count = post_count + 1')
```

Comment:
This is a bug we had. `UserStat.where(user_id: posters)` will only return distinct records. Therefore, we were only changing the count by 1 even if a user posted multiple times in the topic.